### PR TITLE
GH-720 - Update readme regarding timing attacks.

### PR DESCRIPTION
**Pull request by ncb000gt**
Updated the readme with some notes related to timing attacks and the comparison function used in the bcrypt module.

### CODE REVIEW

**Comment by ncb000gt** [comment]
@AndrewCarterUK is this clear enough or do you think it still needs some work?

Also adding for review: @defunctzombie @agathver @jfirebaugh

**Comment by AndrewCarterUK** [comment]
I think I would also remove the `CompareStrings` function, as it now has no benefit over `strcmp` so has no reason to exist.

File: README.md [hunk 1]
```diff
@@ -268,6 +265,16 @@ From @garthk, on a 2GHz core you can roughly expect:
     rounds=31: 2-3 days/hash
 
 
+## A Note on Timing Attacks
+
+Because it's come up multiple times in this project, and other bcrypt projects, it needs to be said- bcrypt comparator functions are not susceptible to timing attacks. They are not susceptible because the comparison function is comparing one hash to another. Basically it boils down to you know nothing about the password you're trying to attack by trying to use a timing attack.
```
Comment (defunctzombie) [inline README.md]: > Basically it boils down to you know nothing about the password you're trying to attack by trying to use a timing attack.

This sentence reads strangely. Not sure what my exact recommendation is other than maybe stating in a sentence or two the _gist_ of the reasoning in bcrypt-ruby#43.

**Comment by defunctzombie** [comment]
> I think I would also remove the `CompareStrings` function, as it now has no benefit over `strcmp` so has no reason to exist.

I agree - tho if it means breaking the API surface then I don't feel as strongly.

**Comment by AndrewCarterUK** [comment]
> > I think I would also remove the `CompareStrings` function, as it now has no benefit over `strcmp` so has no reason to exist.
> 
> I agree - tho if it means breaking the API surface then I don't feel as strongly.

If that's the case then the comment suggsting it is a constant time function could still be removed and it could just proxy through to `strcmp`.

**Comment by ncb000gt** [comment]
I left the API as is but changed it to use strcmp. I also updated the language in the readme.

Thoughts?

**Pull request merged**